### PR TITLE
[wip] make IBD into a MatrixToTableFunction

### PR DESCRIPTION
**Pull request by iitalics**

### CODE REVIEW

File: hail/src/main/scala/is/hail/methods/IBD.scala [hunk 1]
```diff
@@ -304,28 +306,16 @@ object IBD {
...
-
-      val computeMaf = mafFieldName.map(generateComputeMaf(input, _))
-      val sampleIds = input.stringSampleIds
-
-      TableLiteral(TableValue(ctx, ibdPType, FastIndexedSeq("i", "j"),
-        computeIBDMatrix(input, computeMaf, min, max, sampleIds, bounded)), ctx)
+      val ibd = IBD(mafFieldName, bounded, min, max)
+      val mv = Interpret(inputIR, ctx, optimize = true).toMatrixValue(inputIR.typ.colKey)
+      val tv = ibd.execute(ctx, mv)
+      TableLiteral(tv, ctx)
```
Comment (tpoterba) [inline hail/src/main/scala/is/hail/methods/IBD.scala]: The right solution to this change involves deleting `pyApply` entirely -- instead, we would construct a `MatrixToTableApply` IR node in Python containing a serializable representation of the IBD you've written below (which looks totally correct)

Comment (iitalics) [reply hail/src/main/scala/is/hail/methods/IBD.scala]: Gotcha

File: hail/python/hail/methods/statgen.py [hunk 2]
```diff
@@ -105,12 +105,15 @@ def identity_by_descent(dataset, maf=None, bounded=True, min=None, max=None) ->
...
-        joption(max)))
+    dataset = require_biallelic(dataset, 'ibd')
+
+    return Table(MatrixToTableApply(dataset._mir, {
+        'name': 'IBD',
+        'mafFieldName': '__maf' if maf is not None else None,
+        'bounded': bounded,
+        'min': min,
+        'max': max,
+    })).persist()
```
Comment (iitalics) [inline hail/python/hail/methods/statgen.py]: Is `persist()` here correct? I'm not sure what the implications of it are

Comment (tpoterba) [reply hail/python/hail/methods/statgen.py]: persist() is a spark concept that roughly means: "The first time you compute this thing, serialize and store the data in memory and on local disk. The next time you would need to compute it, deserialize these data instead"

I've reviewed the previous semantics and don't see a persist either in Python or Scala, so it's probably safest to leave it off.

However, I'm quite worried about what IBD is doing, so we should add benchmarks and revisit.

Comment (tpoterba) [reply hail/python/hail/methods/statgen.py]: so basically just remove the persist and we're good to go

Comment (iitalics) [reply hail/python/hail/methods/statgen.py]: OK 👍

**Pull request merged**